app/views/projects/show.html.erb

```diff
@@ -1,51 +1,133 @@
+            <tr>
+              <th scope="col">#</th>
+              <th scope="col"><%= t('project.name') %></th>
+              <th scope="col"><%= t('project.email') %></th>
+              <th scope="col"><%= t('project.actions') %></th>
+            </tr>
+          </thead>
+          <tbody>
+            <% @project_general_users.each_with_index do |user, index| %>
+              <tr>
+                <th scope="row"><%= index + 1 %></th>
+                <td><%= user.name %></td>
+                <td><%= user.email %></td>
+                <td>
+                  <%= link_to remove_user_project_path(@project, user_id: user.id), method: :delete, class: 'btn btn-danger btn-sm' do %>
+                    <i class="bi bi-trash"></i>
+                  <% end %>
+                </td>
+              </tr>
+            <% end %>
+          </tbody>
+        </table>
+      </div>
     </div>
-  <% end %>
-<% end %>
+  </div>
 
+  <!-- Modal for Adding Users -->
+  <div class="modal fade" id="addUserModal" tabindex="-1" aria-labelledby="addUserModalLabel" aria-hidden="true">
+    <div class="modal-dialog">
+      <div class="modal-content">
+        <div class="modal-header">
+          <h5 class="modal-title" id="addUserModalLabel"><%= t('project.add_user_to_project') %></h5>
+          <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
+        </div>
+        <div class="modal-body">
+          <% if @organization_general_users.empty? %>
+            <p class="badge bg-secondary"><%= t('project.no_users_exist') %></p>
+          <% else %>
+            <%= form_with url: add_user_project_path(@project), method: :post, local: true do |form| %>
+              <div class="mb-3">
+                <%= form.select 'to_be_added_users', options_from_collection_for_select(@organization_general_users, 'id', 'name'), {}, { multiple: true, class: 'form-select js-multiple-select', style: "width: 100%" } %>
+              </div>
+              <div class="mb-3">
+                <%= form.submit t('project.add_users'), class: 'btn btn-primary' %>
+              </div>
+            <% end %>
+          <% end %>
+        </div>
+      </div>
+    </div>
+  </div>
+
+
+  <!-- Issues List -->
   <%= render 'issues/index', issues: @issues %>
 </div>
+
+
+
+<script>
+  $(document).ready(function() {
+    $('.js-multiple-select').select2({
+      placeholder: "<%= t('project.select_users') %>",
+      allowClear: true,
+      tags: true,
+      width: 'resolve',
+      language: {
+        noResults: function() {
+          return "<%= t('project.no_users_to_show') %>";
+        }
+      },
+      escapeMarkup: function(markup) {
+        return markup;
+      }
+    });
+
+    // Functionality to remove selected users
+    $('.js-multiple-select').on('select2:unselect', function(e) {
+      var unselected_id = e.params.data.id;
+      $(this).find('option[value="'+unselected_id+'"]').remove();
+    });
+  });
+</script>
```

Comment:
can we use js file for this code?

Reply:
also verify in your project this isn't recommended to use js in the html.erb code